File: vm_rental/wwwroot/scripts/countries.js
const countriesRepository = {
      countries: { byIndex: [], byName: [], byCode2: [] },
      
      initialize : function(jsonDataCountries){
        if(jsonDataCountries){        
            for(i = 0; i < jsonDataCountries.length; i++){

                $country = jsonDataCountries[i];    
        
                $countrybyName  = {code2: $country.code2, country_code: $country.country_code, lang: $country.lang, lang_native: $country.lang_native};
                $countrybyCode2 = {name:  $country.name,  country_code: $country.country_code, lang: $country.lang, lang_native: $country.lang_native};

                this.countries.byIndex.push($country);
                this.countries.byName[$country.name] = $countrybyName;
                this.countries.byCode2[$country.code2] = $countrybyCode2;
            }
        }else{ throw "No JSON data Provided for Country Repository."; }
      },
      
      getByIndex : function(){
        return this.countries.byIndex;
      },

      getByName : function(){
        return this.countries.byName;
      },

      getByCode2 : function(){
        return this.countries.byCode2;
      },

      getAll : function(opts){
        if(this.countries !== "undefined" && this.countries.length > 0){
           switch(opts){
              case 'byIndex' : return this.countries.byIndex; break;
              case 'byName'  : return this.countries.byName; break;
              case 'byCode2' : return this.countries.byCode2; break;
           }
        }
        else{throw "No Countries Exist!";}
      },
    };

 //Request the Countries-JSON.
 $jqxhr = $.getJSON("../countries-main.json", function(jsonData){
     countriesRepository.initialize(jsonData.countries);
 });

File: vm_rental/Data/JSON/Countries.cs
using System;
using System.Collections.Generic;
using System.Linq;


namespace vm_rental.Data.JSON
{
  public class Countries
  {
    public List<Country> countries { get; set; }

    public bool CountryExists(string countryName)
    {
      bool countryExists = false;

      if (countryName != null && countries != null)
      {
        countryExists = countries.Exists(c => c.Name.Equals(countryName.Trim(), StringComparison.CurrentCultureIgnoreCase));
      }

      return countryExists;
    }

    public string GetCountryCodeByCountryName(string countryName)
    {
      Country country = null;

      if(countryName != null && countries != null)
      {
        country = countries.Find(c => c.Name.Equals(countryName.Trim(), StringComparison.CurrentCultureIgnoreCase));
      }

      return (country?.Country_code ?? "");
    }
  }
  public class Country
  {
    private string name;
    private string code2;
    private string country_code;
    private string lang;
    private string lang_native;

    public string Name { get => name; set => name = value; }
    public string Code2 { get => code2; set => code2 = value; }
    public string Country_code { get => country_code; set => country_code = value; }
    public string Lang { get => lang; set => lang = value; }
    public string Lang_native { get => lang_native; set => lang_native = value; }
  }
}
